src/main/java/pl/javastart/restoffers/offer/OfferController.java

```diff
@@ -0,0 +1,56 @@
+package pl.javastart.restoffers.offer;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/offers")
+public class OfferController {
+
+    private final OfferService offerService;
+
+    public OfferController(OfferService offerService) {
+        this.offerService = offerService;
+    }
+
+    @GetMapping
+    public List<OfferDto> findFiltered(@RequestParam(required = false) Long categoryId,
+                                       @RequestParam(required = false) String title) {
+
+        return offerService.findFiltered(categoryId, title);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<OfferDto> findById(@PathVariable Long id) {
+        Optional<OfferDto> offerDtoOptional = offerService.findById(id);
+        return offerDtoOptional
+                .map(ResponseEntity::ok)
+                .orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    @GetMapping("/count")
+    public int countOffers() {
+        return offerService.findAllOffers().size();
```

Comment:
Zdefiniuj metodę, która pobierze sam rozmiar z bazy danych. Pobieranie wszystkich rekordów z bazy tylko po to, żeby je policzyć, jest mało wydajne